ddtrace/contrib/django/patch.py

```diff
@@ -485,17 +479,17 @@ def _patch(django):
     if "django.conf.urls.static" not in sys.modules:
         import django.conf.urls.static
 
-    if django.VERSION < (4, 0, 0):
-        trace_utils.wrap(django, "conf.urls.url", traced_urls_path(django))
-
-    if django.VERSION >= (2, 0, 0):
-        trace_utils.wrap(django, "urls.path", traced_urls_path(django))
-        trace_utils.wrap(django, "urls.re_path", traced_urls_path(django))
-
-    # DEV: this check will be replaced with import hooks in the future
+    # Wraps lifecycle methods of class based views
     if "django.views.generic.base" not in sys.modules:
         import django.views.generic.base
-    trace_utils.wrap(django, "views.generic.base.View.as_view", traced_as_view(django))
+    trace_utils.wrap(django, "views.generic.base.View.as_view", traced_view_class(django))
+
+    # A view are wrapt after a URLPattern is resolved
+    # Both URLPattern.resolve() and RegexURLPattern.resolve() return a view
+    urlpattern_resolve = "urls.URLPattern.resolve"
+    if django.VERSION < (2, 0, 0):
+        urlpattern_resolve = "conf.urls.RegexURLPattern.resolve"
+    trace_utils.wrap(django, urlpattern_resolve, traced_resolve_url_pattern(django))
```

Comment:
I added the performance measurements to the PR description. This was the main result: Benchmark hidden because not significant (4): djangosimple-baseline, djangosimple-tracer, djangosimple-profiler, djangosimple-tracer-and-profiler